server.js

```diff
@@ -190,15 +237,11 @@ app.post('/game/:id/player/:pid/move', validate_gid, validate_pid,
     if (chess.turn() == color) {
         var move = chess.move(req.body.move, {sloppy: true}); // need sloppy for algebraic notation
         process_end_game(games[game_id]);
-        if (move) {
-            augment_move(move, games[game_id]);
-            res.status(200).json(move); // send back move
-            games[game_id].last_move = Date.now();
-            console.log(chess.ascii());
-            return;
-        }
-
-        res.status(200).json({error: "invalid move"});
+        // we won't check to see if the move was correct or not. 
+        // Just send back the game state after the move was made.
+        res.status(200).json(games[game_id]); // send back move
```

Comment:
@karadouville How come you got rid of move validation here? Some clients cannot validate moves so we need to have the server validate.

Reply:
@Austinpayne sorry, I was going to talk with you about this and forgot. I didn't realize that some clients would be able to validate the moves and that was necessary for the server. For my purpose, it made sense to have the resulting game state returned. We'll chat more about this tomorrow.

Reply:
I think in this case where you want the game back I would just return it as well with the move object. The nice thing about JSON is you can just add to as you need but replacing here is problematic in this case. Also, since your client already does pre-validation before even sending a move to the server I think we can still leave the "error" case in. Let me know what you think.